src/libraries/System.Private.CoreLib/src/System/String.Searching.cs

```diff
@@ -386,5 +522,119 @@ public int LastIndexOf(string value, int startIndex, int count, StringComparison
+        /// The zero-based index position of <paramref name="value"/> from the end of the current instance
+        /// if that rune is found, or -1 if it is not.
+        /// </returns>
+        public int LastIndexOf(Rune value, StringComparison comparisonType)
+        {
+            return LastIndexOf(value, Length - 1, comparisonType);
+        }
+
+        /// <summary>
+        /// Reports the zero-based index of the last occurrence of the specified rune in the current String object.
+        /// Parameters specify the starting search position in the current string and the type of search to use for
+        /// the specified rune.
+        /// </summary>
+        /// <param name="value">The rune to seek.</param>
+        /// <param name="startIndex">The search starting position. The search proceeds from <paramref name="startIndex"/> toward the beginning of this instance.</param>
+        /// <param name="comparisonType">One of the enumeration values that specifies the rules for the search.</param>
+        /// <returns>
+        /// The zero-based index position of <paramref name="value"/> from the end of the current instance
+        /// if that rune is found, or -1 if it is not.
+        /// </returns>
+        internal int LastIndexOf(Rune value, int startIndex, StringComparison comparisonType)
+        {
+            return LastIndexOf(value, startIndex, startIndex, comparisonType);
+        }
+
+        /// <summary>
+        /// Reports the zero-based index of the last occurrence of the specified rune in the current String object.
+        /// Parameters specify the starting search position in the current string, the number of characters in the
+        /// current string to search, and the type of search to use for the specified rune.
+        /// </summary>
+        /// <param name="value">The rune to seek.</param>
+        /// <param name="startIndex">The search starting position. The search proceeds from <paramref name="startIndex"/> toward the beginning of this instance.</param>
+        /// <param name="count">The number of character positions to examine.</param>
+        /// <param name="comparisonType">One of the enumeration values that specifies the rules for the search.</param>
+        /// <returns>
+        /// The zero-based index position of <paramref name="value"/> from the end of the current instance
+        /// if that rune is found, or -1 if it is not.
+        /// </returns>
+        internal int LastIndexOf(Rune value, int startIndex, int count, StringComparison comparisonType)
+        {
+            ArgumentOutOfRangeException.ThrowIfLessThan(startIndex, 0);
+            ArgumentOutOfRangeException.ThrowIfLessThan(count, 0);
+            ArgumentOutOfRangeException.ThrowIfLessThan(startIndex - count + 1, 0);
+            ArgumentOutOfRangeException.ThrowIfGreaterThan(startIndex, Length - 1);
+
+            int endIndex = startIndex - count;
+
+            for (int index = startIndex; index >= endIndex;)
```

Comment:
Convert Rune to span and then call `public static int LastIndexOf(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, StringComparison comparisonType)`